packages/curve-ui-kit/src/shared/ui/DataTable/TableSearchField.tsx

```diff
@@ -1,27 +1,127 @@
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+import { TableButton } from './TableButton'
 
-export const TableSearchField = ({ value, onChange }: { value: string; onChange: (value: string) => void }) => {
+const { ButtonSize } = SizesAndSpaces
+
+type Props = {
+  value: string
+  onChange: (value: string) => void
+  testId?: string
+  toggleExpanded?: () => void
+  isExpanded: boolean
+}
+
+export const TableSearchField = ({ value, onChange, testId, toggleExpanded, isExpanded = true }: Props) => {
   const searchInputRef = useRef<HTMLInputElement>(null)
   const [isFocused, onFocus, onBlur] = useSwitch()
-  return (
-    <SearchField
-      inputRef={searchInputRef}
-      value={value}
-      onSearch={onChange}
-      size="small"
-      data-testid="table-text-search"
-      onFocus={onFocus}
-      onBlur={onBlur}
-      sx={(t) => ({
+  const isMobile = useIsMobile()
+  const collapsible = !!toggleExpanded
+
+  const handleExpand = useCallback(() => {
+    toggleExpanded?.()
+    // Focus the input when expanding
+    const timeout = setTimeout(() => {
+      searchInputRef.current?.focus()
+    }, Duration.Focus)
+    return () => clearTimeout(timeout)
+  }, [toggleExpanded])
+
+  const handleBlur = useCallback(() => {
+    onBlur()
+    // Collapse when unfocused and empty in collapsible mode
+    if (collapsible && !value) {
+      toggleExpanded?.()
+    }
+  }, [onBlur, value, collapsible, toggleExpanded])
+
+  return collapsible ? (
+    <Box
+      sx={{
+        display: 'flex',
+        // on mobile when search is de-expanded, animation doesn't look good
+        transition: isMobile && !isExpanded ? 'none' : `${TransitionFunction}`,
+        flex: isExpanded ? '1 1 auto' : `0 0 ${ButtonSize.sm}`,
+        width: isExpanded ? 0 : `${ButtonSize.sm}`,
+        overflow: 'hidden',
+      }}
+    >
+      {isExpanded ? (
+        <SearchFieldWithTransition
+          value={value}
+          searchInputRef={searchInputRef}
+          onChange={onChange}
+          onFocus={onFocus}
+          onBlur={handleBlur}
+          isFocused={isFocused}
+          testId={testId}
+          sx={{ flex: '1 1 auto', minWidth: 0 }}
+        />
+      ) : (
+        <TableButton
+          onClick={handleExpand}
+          icon={SearchIcon}
+          active={isExpanded}
+          testId={testId ? `btn-expand-search-${testId}` : 'btn-expand-search'}
+        />
+      )}
+    </Box>
+  ) : (
+    <Box
+      sx={{
+        width: { mobile: '100%', tablet: '478px' },
+        maxWidth: '100%',
+      }}
+    >
+      <SearchFieldWithTransition
+        value={value}
+        searchInputRef={searchInputRef}
+        onChange={onChange}
+        onFocus={onFocus}
+        onBlur={onBlur}
+        isFocused={isFocused}
+        testId={testId}
+        sx={{ flex: '1 1 auto', minWidth: 0 }}
+      />
+    </Box>
+  )
+}
+
+const SearchFieldWithTransition = ({
```

Comment:
I've asked this before, the name is improved but is it really necessary? Can't we just use a variable?